File: CommunicationModel/src/BrokerModels/SensorReaderEvent.cs
using System.Collections.Generic;

namespace CommunicationModel.BrokerModels
{

	public class SensorReaderEvent : ServiceEvent
	{
		public string SensorName { get; set; }
		public int LastReadIndex { get; set; }

		public string IpAddress { get; set; }
		public int ListeningPort { get; set; }

		public List<string> DataHeader { get; set; }
		public SensorValues NewData { get; set; }

		public SensorReaderEvent(
			string serviceId,
			string sensorName,
			List<string> dataHeader,
			int lastReadIndex,
			SensorValues newData,
			string ipAddress,
			int listeningPort,
			string customMessage = "")
			
			: base(serviceId, ServiceType.SensorReader, customMessage)
		{
			SensorName = sensorName;
			LastReadIndex = lastReadIndex;

			IpAddress = ipAddress;
			ListeningPort = listeningPort;
			DataHeader = dataHeader;
			NewData = newData;
		}
	}

}

File: DashboardClient/TBoardRabbitAdapter/src/main/java/broker/messages/incoming/SensorReaderEvent.java
package broker.messages.incoming;

import java.util.List;

public class SensorReaderEvent extends ServiceEvent {

	public String SensorName;
	public int LastReadIndex;

	public String IpAddress;
	public int ListeningPort;

	public List<String> DataHeader;
	public SensorValues NewData;

	public SensorReaderEvent(
			String sourceId,
			ServiceType type,
			String message,
			String SensorName,
			int LastReadIndex,
			String IpAddress,
			int ListeningPort,
			List<String> DataHeader,
			SensorValues NewData) {
		super(sourceId, type, message);

		this.SensorName = SensorName;
		this.LastReadIndex = LastReadIndex;
		this.IpAddress = IpAddress;
		this.ListeningPort = ListeningPort;
		this.DataHeader = DataHeader;
		this.NewData = NewData;
	}

}
